src/home_robot/home_robot/mapping/voxel.py

```diff
@@ -25,37 +30,177 @@ def combine_point_clouds(
+        self._disk_size = np.ceil(1.0 / self.grid_resolution)
+        self._visited_disk = create_disk(
+            1.0 / self.grid_resolution, (2 * self._disk_size) + 1
+        )
+
+        # Add points with local_radius to the voxel map at (0,0,0) unless we receive lidar points
+        self.add_local_radius_points = add_local_radius_points
+        self.local_radius = local_radius
+        # TODO:
+        # Cast lines to the center from nearest points - this is to create explored re
+        if cast_to_center is True:
+            raise NotImplementedError("cast to center not implemented")
+
+        if grid_size is not None:
+            self.grid_size = [grid_size[0], grid_size[1]]
+        else:
+            self.grid_size = DEFAULT_GRID_SIZE
+        # Track the center of the grid - (0, 0) in our coordinate system
+        # We then just need to update everything when we want to track obstacles
+        self.grid_origin = np.array(self.grid_size + [0]) // 2
+        # Create map here
+        self.reset_cache()
+
+    def reset_cache(self):
+        """Clear some tracked things"""
+
+        # Stores points in 2d coords where robot has been
+        self._visited = np.zeros(self.grid_size)
+
+        # Store instances detected (all of them for now)
+        self._instance_views = []
+
+        # Store 2d map information
+        # This is computed from our various point clouds
+        self._map2d = None
+
+        # Holds 3d data
+        self.xyz, self.feats = None, None
+
+    def add(
+        self,
+        camera_pose: np.ndarray,
+        xyz: np.ndarray,
+        feats: np.ndarray,
+        depth: Optional[np.ndarray] = None,
+        base_pose: Optional[np.ndarray] = None,
+        obs: Optional[Observations] = None,
+        **info
+    ):
+        """Add this to our history of observations. Also update the current running map.
+
+        Parameters:
+            camera_pose(np.ndarray): necessary for measuring where the recording was taken
+            xyz: N x 3 point cloud points
+            feats: N x D point cloud features; D == 3 for RGB is most common
+            base_pose: optional location of robot base
+            obs: observations
+        """
+        # TODO: we should remove the xyz/feats maybe? just use observations as input?
+        # TODO: switch to using just Obs struct?
         assert xyz.shape[-1] == 3
         assert feats.shape[-1] == self.feature_dim
         assert xyz.shape[0] == feats.shape[0]
-        self.observations.append((camera_pose, xyz, feats, info))
+
+        # apply depth filter
+        feats = feats.reshape(-1, 3)
+        xyz = xyz.reshape(-1, 3)
+
+        if depth is not None:
+            depth = depth.reshape(-1)
+            valid_depth = np.bitwise_and(depth > self.min_depth, depth < self.max_depth)
+            feats = feats[valid_depth, :]
+            xyz = xyz[valid_depth, :]
+        else:
+            valid_depth = np.ones_like(xyz)
+
+        if len(xyz.shape) > 2:
```

Comment:
Is this condition ever going to be true? Line 153-154 already do the reshaping and the lines between 156-162 aren't going to change the shapes of those arrays

Reply:
removed